{
  try {
    final int segmentSize=52;
    MemorySegment seg=createSegment(segmentSize);
    byte[] bytes=new byte[segmentSize];
    random.nextBytes(bytes);
    seg.put(0,bytes);
{
      DataOutputStream out=new DataOutputStream(new ByteArrayOutputStream());
      try {
        seg.get(out,-1,segmentSize / 2);
        fail("IndexOutOfBoundsException expected");
      }
 catch (      Exception e) {
        assertTrue(e instanceof IndexOutOfBoundsException);
      }
      try {
        seg.get(out,segmentSize,segmentSize / 2);
        fail("IndexOutOfBoundsException expected");
      }
 catch (      Exception e) {
        assertTrue(e instanceof IndexOutOfBoundsException);
      }
      try {
        seg.get(out,-segmentSize,segmentSize / 2);
        fail("IndexOutOfBoundsException expected");
      }
 catch (      Exception e) {
        assertTrue(e instanceof IndexOutOfBoundsException);
      }
      try {
        seg.get(out,Integer.MIN_VALUE,segmentSize / 2);
        fail("IndexOutOfBoundsException expected");
      }
 catch (      Exception e) {
        assertTrue(e instanceof IndexOutOfBoundsException);
      }
      try {
        seg.get(out,Integer.MAX_VALUE,segmentSize / 2);
        fail("IndexOutOfBoundsException expected");
      }
 catch (      Exception e) {
        assertTrue(e instanceof IndexOutOfBoundsException);
      }
    }
{
      DataInputStream in=new DataInputStream(new ByteArrayInputStream(new byte[segmentSize]));
      try {
        seg.put(in,-1,segmentSize / 2);
        fail("IndexOutOfBoundsException expected");
      }
 catch (      Exception e) {
        assertTrue(e instanceof IndexOutOfBoundsException);
      }
      try {
        seg.put(in,segmentSize,segmentSize / 2);
        fail("IndexOutOfBoundsException expected");
      }
 catch (      Exception e) {
        assertTrue(e instanceof IndexOutOfBoundsException);
      }
      try {
        seg.put(in,-segmentSize,segmentSize / 2);
        fail("IndexOutOfBoundsException expected");
      }
 catch (      Exception e) {
        assertTrue(e instanceof IndexOutOfBoundsException);
      }
      try {
        seg.put(in,Integer.MIN_VALUE,segmentSize / 2);
        fail("IndexOutOfBoundsException expected");
      }
 catch (      Exception e) {
        assertTrue(e instanceof IndexOutOfBoundsException);
      }
      try {
        seg.put(in,Integer.MAX_VALUE,segmentSize / 2);
        fail("IndexOutOfBoundsException expected");
      }
 catch (      Exception e) {
        assertTrue(e instanceof IndexOutOfBoundsException);
      }
    }
  }
 catch (  Exception e) {
    e.printStackTrace();
    fail(e.getMessage());
  }
}
